CONTRIBUTING.md

```diff
@@ -0,0 +1,65 @@
+# Contributing
+
+We are open to, and grateful for, any contributions made by the community. By contributing to Redux Toolkit, you agree to abide by the [code of conduct](https://github.com/reduxjs/redux-toolkit/blob/master/CODE_OF_CONDUCT.md).
+
+## Reporting Issues and Asking Questions
+
+Before opening an issue, please search the [issue tracker](https://github.com/reduxjs/redux-toolkit/issues) to make sure your issue hasn't already been reported.
+
+Please ask any general and implementation specific questions on [Stack Overflow with a Redux tag](http://stackoverflow.com/questions/tagged/redux?sort=votes&pageSize=50) for support.
+
+## Development
+
+Visit the [Issue tracker](https://github.com/reduxjs/redux-toolkit/issues) to find a list of open issues that need attention.
+
+Fork, then clone the repo:
+
+```
+git clone https://github.com/your-username/redux-toolkit.git
+```
+
+### Building
+
+Running the `build` task will create both a CommonJS module-per-module build and a UMD build.
+
+```
+npm run build
+```
+
+### Testing and Linting
+
+To run the tests:
+
+```
+npm run test
+```
+
+To continuously watch and run tests, run the following:
+
+```
+npm test -- --watch
+```
+
+To perform linting with `eslint`, run the following:
+
+```
+npm run lint
+```
+
+### New Features
+
+Please open an issue with a proposal for a new feature or refactoring before starting on the work. We don't want you to waste your efforts on a pull request that we won't want to accept.
+
+## Submitting Changes
+
+- Open a new issue in the [Issue tracker](https://github.com/reduxjs/redux-toolkit/issues).
+- Fork the repo.
+- Create a new feature branch based off the `master` branch.
+- Make sure all tests pass and there are no linting errors.
+- Submit a pull request, referencing any issues it addresses.
```

Comment:
- If you changed external-facing types, make sure to also build the project locally and include the updated [etc/redux-toolkit.api.md](https://github.com/reduxjs/redux-toolkit/blob/master/etc/redux-toolkit.api.md) in your pull request.